pkg/runtime/runtime_test.go

```diff
@@ -1577,6 +1570,13 @@ func TestOnNewPublishedMessageGRPC(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			port, _ := freeport.GetFreePort()
+			rt := NewTestDaprRuntimeWithProtocol(modes.StandaloneMode, string(GRPCProtocol), port)
```

Comment:
does `NewTestDaprRuntimeWithProtocol` spin up any http or grpc servers ? please ensure that the test will shutdown the servers after test is done.

Reply:
NewTestDaprRuntimeWithProtocol is same as NewTestDaprRuntime just with ability to specify GRPCProtocol and port. 

The call to create local app channel has a following defer closing the app client along with the defer for gRPCServer stop.